### PR TITLE
Generate 3.6 wheels

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ python:
   - 2.7
   - 3.4
   - 3.5
+  - 3.6
```
Comment (charris) [inline .travis.yml]: You want `3.6-dev` here.

File: .travis.yml [hunk 2]
```diff
@@ -65,6 +66,8 @@ matrix:
       env: USE_WHEEL=1 RUN_FULL_TESTS=1
     - python: 3.5
       env: USE_SDIST=1
+    - python: 3.6
```
Comment (charris) [inline .travis.yml]: Maybe here also.

**Comment by graingert** [comment]
3.6 is available in yakkety so it's probably worth creating wheels for 3.6

**Comment by charris** [comment]
Looks like some tests will need fixing. When this gets ready, the commits should be squashed and and follow the commit message template in `doc/source/dev/gitwash/development_workflow.rst`. We could maybe start by omitting the wheel generation until things are fixed in the normal test matrix.

**Comment by njsmith** [comment]
Testing on 3.6 is a good idea, but we shouldn't distribute wheels yet -- the ABI isn't finalized until the final release.

**Comment by njsmith** [comment]
(Meaning, if we build wheels now, there's every chance that they'll start segfaulting as soon as the next 3.6 beta comes out.)

**Comment by graingert** [comment]
Python3.6 has been 'released' with Ubuntu Yakkety so people will be ready
to pip install stuff pretty soon.

I imagine most uses on 3.6 right now will be ready to pip install a new
version after the ABI freezes.

Thomas Grainger

**Comment by njsmith** [comment]
(Sorry, fat-fingered the button there.)

It looks like yakkety is still shipping with 3.5 as the default python3: http://packages.ubuntu.com/yakkety/python3

And yeah, of course once 3.6-final is out then we should start shipping wheels for it.

**Comment by graingert** [comment]
sure but 3.6 is available to install now, users will be expecting to install numpy 3.6 before the ABI is final.

**Comment by njsmith** [comment]
Well, they can install it, they just have to build it themselves. Like the bad old days of last year :-)

The problem is that if we put up a wheel on pypi marked as being for "py36", then pip will happily install that on any version of py36, including the future ones where we don't know the ABI yet. And then we have to deal with the fallout if suddenly lots of people are getting a segfaulting numpy.

**Comment by charris** [comment]
@graingert I've added 3.6-dev testing along with the fix for the Python 3.6 `ResourceWarning` in #8178. I'd like to wait to generate wheels until the official 3.6 release, which looks to be in December, so closing this. Thanks for getting the ball rolling here.

**Comment by tdsmith** [comment]
Do you have a sense of when we should expect py36 wheels? (Sorry to nag; happy holidays to all celebrating!)

**Comment by charris** [comment]
I'm considering making them for NumPy 1.12.0 now that Python 3.6 is out.

**Comment by ogrisel** [comment]
@matthew-brett used to generate wheels for all releases of numpy, scipy & co in the past. I think this is a good idea (assuming tests failures only happen for trivial changes such as dict ordering or changes in  exception messages).

**Comment by ogrisel** [comment]
FYI we have updated the manylinux docker images to include Python 3.6.0.

**Pull request closed without merging**